pkg/sql/pgwire/conn_test.go

```diff
@@ -1580,8 +1579,10 @@ func TestCancelQuery(t *testing.T) {
 	require.NoError(t, err)
 	defer db.Close()
 
-	// Cancellation has no effect on ongoing query.
-	if _, err := db.QueryContext(cancelCtx, "select pg_sleep(0)"); err != nil {
+	// Cancellation should stop the query.
+	if _, err := db.QueryContext(cancelCtx, "select pg_sleep(1)"); err == nil {
```

Comment:
i would sleep longer (30s?) in case of, e.g., race.

Reply:
sounds good